pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -47,8 +49,8 @@ def __init__(self, min_steps: int, max_steps: int):
         self.batches_seen: int = 0
         self.is_last_batch: Optional[bool] = None
 
-        self.batch_loop: Optional[TrainingBatchLoop] = None
-        self.val_loop: Optional[loops.EvaluationLoop] = None
+        self.batch_loop = TrainingBatchLoop()
+        self.val_loop = loops.EvaluationLoop()
```

Comment:
I like this, but means we should remove the creation in `connect`

And `connect` should take an optional argument for the user to pass a custom loop

`val_loop: Optional[Loop] = None`

and then

`self.val_loop = val_loop or self.val_loop`

Reply:
Let's take care of this when actually getting into customisation.

Reply:
We can wait for the `loop` argument but the creation in `connect` should be removed in this PR

Reply:
Done !